plans/external.pp

```diff
@@ -0,0 +1,42 @@
+# @summary
+#   A plan that generates external facts based on the provided modulepath and
+#   sets facts on specified targets.
+#
+# @param path The path to the directory on localhost containing external facts
+# @param targets The targest the collect and set facts on
+# @return The target objects, with facts added
+# @example Gather external facts from an installed module
+#   $moduledir = module_directory('mymod')
+#   $with_facts = run_plan(facts::external, $targets, path => file::join($moduledir, 'facts.d'))
+#   return $with_facts.map |$target| { $target.facts }
+plan facts::external(
+  String $path,
+  TargetSpec $targets
+) {
+  $t = get_targets($targets)
+
+  # Strip out dotfiles
+  $facts_paths = dir::children($path).filter |$path| { $path[0] != '.' }
+  $facts_paths.each |$file| {
+    $ext_facts = run_script(file::join($path, $file), $t)
+    $t.each |$target| {
+      $raw_facts = $ext_facts.find($target.name).value['stdout']
+      $jsonoryaml = catch_errors() || {
+        parseyaml($raw_facts)
+      }
+
+      # If parsing as YAML or JSON failed, the error will be a string and we
+      # should try to parse the original facts as key-value pairs
+      if type($jsonoryaml, 'generalized') == String {
+        # Split on newlines and equals, then convert to a hash
+        $facts_for_target = Hash.new($raw_facts.split(/\n|\=/))
```

Comment:
We should split the string into lines and then separately split each line on `=`. This will currently accept output like this:

```
key
val
key2
val2
```

which isn't valid.

This will also incorrectly parse if the fact _value_ contains an `=`.